engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -177,14 +177,24 @@ struct mind_blast_t final : public priest_spell_t
         {
           pet->adjust_duration( your_shadow_duration );
         }
+        else if ( priest().t28_4pc_summon_event &&
```

Comment:
Hmm I don't like the remains check here. Please remove it but set the event pointer to null after you spawn the pet.

You can't assume that just because remains returns 0 that the event was already executed.